### ISSUE TITLE
Could loss function output a vector instead of a scalar ?

### ISSUE THREAD

**Issue by YujieDengJames** [pytorch/pytorch#2147]
In online hard example mining, it is helpful to know the loss for each instance separately.

**Comment by fmassa** [comment]
Duplicate of https://github.com/pytorch/pytorch/issues/264

**Comment by YujieDengJames** [comment]
The problem is that I want to compute the loss for each instance, and sort the losses. The OHEM will only keep the high loss instances for backpropagation. What you refer to is adding weights to loss vector. It is not obvious to me how to make this transition .

**Comment by Jiaming-Liu** [comment]
Outputing a loss vector seems to be an elegant solution for all the related issues. The current "weighted" method are limited to the specific use cases. Although one can easily re-implement the loss module using tensor operations, it is not trivial to achieve the efficiency of the built-in loss functions.

**Comment by fmassa** [comment]
@YujieDengJames The issue I referenced is exactly what you are asking for.
Instead of summing the elements on the mini-batch, having the ability of returning each one of them separately. Something like
```python
def mse_loss(x, y, size_average=False, per_element=False):
    loss_per_element = (x - y) ** 2
    if per_element:
        return loss_per_element
    if size_average:
        return loss_per_element.mean()
    return loss_per_element.sum()
```

**Comment by apaszke** [comment]
As @fmassa noted, it's a duplicate. Let's use that issue to track this.

**Comment by s-basant** [comment]
I am sorry if my question is very silly but I am confused as I am new to this, if possible please help me to understand for below scenario.

I have time-series data as below
time(t1 to t6)
time  data
t1  [1,2,3,4]
t2  [2,3,4,5]
t3  [3,4,5,6]
t4  [4,5,6,7]
t5  [5,6,7,8]
t6  [6,7,8,9]

if the window =  2, 
I can create the training dataset below.

input -> output
[[1,2,3,4] ,[2,3,4,5] ] ->[3,4,5,6]
[[2,3,4,5], [3,4,5,6] ] ->[4,5,6,7]
[[3,4,5,6]  [4,5,6,7] ] ->[5,6,7,8]
[[4,5,6,7], [5,6,7,8]] -> [6,7,8,9]

now in this scenario how to calculate the loss.. ??
